modules/yieldbotBidAdapter.js

```diff
@@ -0,0 +1,806 @@
+   * @property {string} REQUEST_PARAMS.LANGUAGE Language and locale of the user-agent
+   * @property {string} REQUEST_PARAMS.NAVIGATION_PLATFORM User-agent browsing platform
+   * @property {string} REQUEST_PARAMS.USER_AGENT User-Agent string
+   * @property {string} [REQUEST_PARAMS.LAST_PAGEVIEW_TIME] Time in milliseconds since the last page visit
+   * @property {string} REQUEST_PARAMS.NAVIGATION_START_TIME Performance timing navigationStart
+   * @property {string} REQUEST_PARAMS.ADAPTER_LOADED_TIME Adapter code interpreting started timestamp, in milliseconds since the UNIX epoch
+   * @property {string} REQUEST_PARAMS.BID_REQUEST_TIME Yieldbot bid request sent timestamp, in milliseconds since the UNIX epoch
+   * @property {string} REQUEST_PARAMS.BID_RESPONSE_TIME Yieldbot bid response processing started timestamp, in milliseconds since the UNIX epoch
+   * @property {string} REQUEST_PARAMS.AD_REQUEST_TIME Yieldbot ad creative request sent timestamp, in milliseconds since the UNIX epoch
+   * @property {string} REQUEST_PARAMS.AD_RENDER_TIME Yieldbot ad creative render started timestamp, in milliseconds since the UNIX epoch
+   * @property {string} REQUEST_PARAMS.AD_IMPRESSION_TIME Yieldbot ad impression rerquest sent  timestamp, in milliseconds since the UNIX epoch
+   * @property {string} [REQUEST_PARAMS.INTERSECTION_OBSERVER_AVAILABLE] Indicator that the user-agent supports the Intersection Observer API
+   * @property {string} [REQUEST_PARAMS.IFRAME_TYPE] Indicator to specify Yieldbot creative rendering occured in a same (<code>so</code>) or cross (<code>co</code>) origin iFrame
+   * @property {string} [REQUEST_PARAMS.BID_TYPE] Yieldbot bid request type: initial or refresh
+   * @property {string} REQUEST_PARAMS.CALLBACK Ad creative render callback
+   * @property {string} REQUEST_PARAMS.SESSION_BLOCKED Yieldbot ads blocked by user opt-out or suspicious activity detected during session
+   * @property {string} [REQUEST_PARAMS.ADAPTER_ERROR] Yieldbot error description parameter
+   * @property {string} [REQUEST_PARAMS.TERMINATOR] Yieldbot search parameters terminator
+   * @property {object} COOKIES Cookie name suffixes set by Yieldbot. See also <code>YieldbotAdapter._COOKIE_PREFIX</code>
+   * @property {string} COOKIES.SESSION_BLOCKED The user session is blocked for bids
+   * @property {string} COOKIES.SESSION_ID The user session identifier
+   * @property {string} COOKIES.PAGEVIEW_DEPTH The session pageview depth
+   * @property {string} COOKIES.USER_ID The Yieldbot first-party user identifier
+   * @property {string} COOKIES.LAST_PAGEVIEW_ID The last pageview identifier within the session TTL
+   * @property {string} COOKIES.PREVIOUS_VISIT The time in [ms] since the last visit within the session TTL
+   * @property {string} COOKIES.URL_PREFIX Geo/IP proximity request Url domain
+   * @TODO Document parameter optionality for properties
+   */
+  CONSTANTS: {
```

Comment:
Just noting that production build does not mangle object properties. Structures like this `CONSTANTS` object will contribute a reasonable amount to module size.

The general intent of having such an object is for understanding and reasoning adapter behavior. We expect more and more publisher integration via the Prebid.org channel and believe clarity and transparency in the adapter is a sound goal.

Reply:
Regarding `CONSTANTS` and module file size, clearly this too could be optimized manually. Fewer than expected static strings are used more than once so these could be inline.

With using `CONSTANTS` explicitly it does co-locate the jsdoc with more context. We could keep the request parameter documentation in the `yieldbotBidAdapter.md` for general documentation and cut down on the jsdoc if we do go with more static strings in place.

Anyway, I'm open to make changes here if the project (@mkendall07) feels it better practice to have developers optimize for this. Are there any plans to have have uglify `{mangle: { properties: true } }`?

Reply:
It's a good point, we were discussing this internally the other day. I was suggesting we investigate other options such as closure-compiler (now that it's been ported to JS) and evaluate from there. We are definitely interested in further optimizations of the build, with that said you could use `const` for these if you wanted. The only reason to use a `constant` type object would be if you needed it in multiple files I think?

Reply:
@mkendall07 nice to hear about the closure-compiler potential direction. It looks like in closure-compiler you'd have to have on `ADVANCED_OPTIMIZATIONS` for the constants to be dealt with completely. I'd imagine that getting closure-compiler ADVANCED_OPTIMIZATIONS to work will be more involved than getting uglify to have mangling on. I'm a fan of whatever approach for long term sustainability of the project seems to make good sense and this adapter should fit in with where the whole project is generally headed. There's no concern here from me that should block this getting merged solely because the build may be headed towards a different way to be optimized.